### ISSUE TITLE
rejectUnauthorized does not work in @gitbeaker/node

### ISSUE THREAD

**Issue by JeremyDurnell** [jdalrymple/gitbeaker#1250]
**Description**

Specifying `rejectUnauthorized: false` does nothing in node

**Steps to reproduce**

1. set `rejectUnauthorized` to false
2. request something

**Expected behaviour**
Requested resource should be retrieved even if the server has an invalid ssl cert

**Actual behaviour**
Server certificate will be validated and, in the case of an invalid cert, an error will be thrown

**Possible fixes**

change:
```javascript
if (service.url.includes('https') && service.rejectUnauthorized) {
    options.https = {
      rejectUnauthorized: service.rejectUnauthorized,
    };
  }
```
to
```javascript
if (service.url.includes('https')) {
    options.https = {
      rejectUnauthorized: service.rejectUnauthorized,
    };
  }
```
https://github.com/jdalrymple/gitbeaker/blob/master/packages/gitbeaker-node/src/GotRequester.ts#L31

**Comment by jdalrymple** [comment]
Oh wow, thats a silly check. Should of been checking if its not null!

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`25.1.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/25.1.0) :rocket:

**Comment by uncaught** [comment]
@jdalrymple should this option work with self-signed certificates?

I'm still getting this error, even with `rejectUnauthorized: false`:

```
node:internal/deps/undici/undici:13179
      Error.captureStackTrace(err);
            ^

TypeError: fetch failed
    at node:internal/deps/undici/undici:13179:13
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async defaultRequestHandler (file:///app/node_modules/@gitbeaker/rest/dist/index.mjs:66:22)
    at async file:///app/node_modules/@gitbeaker/core/dist/index.mjs:99:22
    at async saveProject (/app/src/projects.mts:14:20)
    at async file:///app/save-project:16:1 {
  [cause]: Error: unable to verify the first certificate
      at TLSSocket.onConnectSecure (node:_tls_wrap:1679:34)
      at TLSSocket.emit (node:events:520:28)
      at TLSSocket._finishInit (node:_tls_wrap:1078:8)
      at ssl.onhandshakedone (node:_tls_wrap:864:12) {
    code: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE'
  }
}

Node.js v22.7.0
```

@gitbeaker/rest: 40.1.3

-----

Adding `process.env['NODE_TLS_REJECT_UNAUTHORIZED'] = '0';` works though.
